src/Omega_h_vector.hpp

```diff
@@ -150,6 +150,16 @@ OMEGA_H_INLINE Vector<n> zero_vector() OMEGA_H_NOEXCEPT {
   return fill_vector<n>(0.0);
 }
 
+template <Int n>
+OMEGA_H_INLINE Vector<n> max_vector() OMEGA_H_NOEXCEPT {
+  return fill_vector<n>(Kokkos::Experimental::finite_max_v<Real>);
```

Comment:
The CI build with kokkos disabled failed (a serial CPU build).  Here is the cmake command that was used:
https://github.com/SCOREC/omega_h/actions/runs/9932729889/job/27455899042?pr=108#step:8:1

A couple solutions come to mind:
1. add `#if defined(OMEGA_H_USE_KOKKOS) .... #else .... #endif` around the existing code and add the non-kokkos implementation in the `#else` block.
2. remove the helper functions (they aren't used anywhere else) and call the needed code in `bboxWrap()` directly.

The first isn't great as `Omega_h_vector.hpp` doesn't have any direct kokkos `ifdef` blocks.  The second isn't ideal either, but would be quick and somewhat cleaner (less changes).

Reply:
The current wrapper functions have a slightly wrong scope. The wrapper functions should provide the max/min values that can be passed into `fill_vector` and hide the ifdefs.

Since we only use this here for now, let's directly use the `#ifdef`. A more future proof/robust option would be to have our own header like so where we return either std or kokkos based on if we compile with kokkos support / are running on the GPU.

One additional aspect I'm not sure about is if the `std::numeric_limits` functions compile on the gpu.

Reply:
> Since we only use this here for now, let's directly use the #ifdef. 

This is option 2 correct?

Reply:
I have removed the `max_vector` and `min_vector` from `Omega_h_vector.hpp` and replaced them with the `fill_vector` in the `bboxWrap`. Now, it's compiling and passing all the tests with the configuration you pointed above.